discord/channel.py

```diff
@@ -624,6 +624,24 @@ def comparator(channel):
         ret.sort(key=comparator)
         return ret
 
+    @property
+    def text_channels(self):
+        """List[:class:`TextChannel`]: Returns the text channels that are under this category."""
+        ret = [c for c in self.guild.channels
+            if c.category_id == self.id
+            and isinstance(c, TextChannel)]
+        ret.sort(key=lambda c: c.position)
```

Comment:
Comparator should take into consideration both ID and the position as seen in other channel/role sorters.